packages/app/components/Schedule/UpdateCalendarButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { API } from "@koh/api-client";
+import { Role } from "@koh/common";
+import { Button } from "antd";
+import React, { ReactElement, useState } from "react";
+import { useRoleInCourse } from "../../hooks/useRoleInCourse";
+
+enum CalendarUpdateStatus {
+  BEFORE,
+  UPDATING,
+  AFTER,
+}
+
+type UpdateCalendarButtonProps = {
+  courseId: number;
+};
+
+export default function UpdateCalendarButton({
+  courseId,
+}: UpdateCalendarButtonProps): ReactElement {
+  const [updated, setUpdated] = useState(CalendarUpdateStatus.BEFORE);
+  const role = useRoleInCourse(courseId);
+
+  const updateCalendar = async () => {
+    setUpdated(CalendarUpdateStatus.UPDATING);
+    await API.course.updateCalendar(courseId);
+    setUpdated(CalendarUpdateStatus.AFTER);
```

Comment:
Will the calendar actually update after this happens? Or do we need to refetch the `officeHours` for a course to make that happen?

Reply:
it updated the calendar when I updated on local :o